client/policy.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package client
+
+// PolicyDescription describes a policy which is made up of a valid policyID that is
+// registered with acp module and has a valid DPI compliant resource name that also
+// exists on that policy, the description is already validated.
+type PolicyDescription struct {
+	// ID is the policyID that is unique to every policy.
+	ID string
```

Comment:
question: Why does this need to exist, and why can we not just have `ResourceName` on `CollectionDescription`?

Reply:
You would rather have a `ResourceName` and `PolicyID` on `CollectionDescription` than this struct? Why mix the `@policy` specified stuff with the rest?

Reply:
Sorry, I just found where this was used - this is an externally managed ID?  It needs to be unique within the ACPModule (i.e. SourceHub) at somepoint?

todo: If that is the case can you please document that here, as at the moment it just looks like a local Id, similar to Collection, Field and Index IDs.

Reply:
> Sorry, I just found where this was used - this is an externally managed ID? It needs to be unique within the ACPModule (i.e. SourceHub) at somepoint?
> 
> todo: If that is the case can you please document that here, as at the moment it just looks like a local Id, similar to Collection, Field and Index IDs.

Correct, this is the PolicyID from the `@policy(id:"", resource:"aa")` on the schema that was parsed in. Where that policyID was grants after AddPolicy to acp module succeeded.

Would you rather I change it to `PolicyID` instead of `ID` and add that documentation?

- [ ] Document On ID it's externally managed through ACP Module

- [ ] Maybe rename ID Variable

Reply:
I'm happy with the current name, Policy.PolicyID is a bit odd and not really consistent with the rest of our public ID prop names.